### ISSUE TITLE
[bug] LSP in 0.26.0 does not work if config file is outside the workspace

### ISSUE THREAD

**Issue by kam193** [ast-grep/ast-grep#1402]
This is the same problem as #1326, so I copied part of the description :D 

# ⏯ Playground Link

[Playground link with relevant code](https://ast-grep.github.io/playground.html)

# 💻 Code

Code:

```
exec("something")
```

Pattern:

```yml
id: example
message: Using exec in code
severity: error
language: Python
rule:
  pattern: exec($CODE)
```

Configuration: I use the LSP pragmatically in a project, but to verify the problem, I configured a similar environment in the VSCode using the official extension.

Directory stucture is like:
```
| rules/
| | sgconfig.yaml
| | python/
| | | rules.yaml
| src/
| | code.py
```

LSP was configured with the absolute path to the `sgconfig.yaml`.

# 🙁 Actual behavior

After updating to 0.26.0 (from PyPI), the LSP server does not work if the config file is not in the workspace root. Rules are loaded, but no diagnostic notes are provided. My suspicion is that this time fix for #1382 caused ignoring files that are not relative to the _config_, instead of _workspace_.

While ignoring files outside the workspace sounds expected, the localisation of the config should not play any role (e.g. you can have a one central place you keep your ASTGrep configuration used in multiple workspaces). LSP protocol sends the current workspace root, and this should be used. I'm not sure (I haven't found the right place in code) why it looks like the workspace depends on the config location.

Old screens (from previous issue), but I've checked the same scenario with 0.26.0, it looks exactly the same: 
![obraz](https://github.com/user-attachments/assets/61618b07-fc16-479a-94d8-457e4b759c93)
![obraz](https://github.com/user-attachments/assets/cc5b9549-6f29-4239-bb91-8d8b5fa48250)

# 🙂 Expected behavior

If using the version >0.25.0 and <0.26.0 (from PyPI) **or** moving `sgconfig.yaml` to the workspace root (adjusting paths to rules), everything works as expected:

![obraz](https://github.com/user-attachments/assets/c5a2e77c-9a1c-4d5e-825c-a6e9010f2d36)

**Comment by HerringtonDarkholme** [comment]
Yeah... Indeed it should be workspace path instead of configuration path. Thanks for the report

**Comment by HerringtonDarkholme** [comment]
fixed in 0.26.2
